### DISCUSSION TITLE
What does SYSTEM_PRIVATE_CORELIB mean?

### DISCUSSION THREAD

**Discussion by MisinformedDNA** [dotnet/runtime#86159] [Q&A] [answered]
In the source code, when SYSTEM_PRIVATE_CORELIB is used, what is the scenario for it being true and what is the scenario for it being false? What state is the symbol in when building a NuGet package?

**Comment by stephentoub** [comment, marked as answer]
Sometimes a source file is compiled into multiple assemblies, e.g. there's some public type that's built into corelib but then some assembly needs its own internal copy of that file.  The Corelib project defines the SYSTEM_PRIVATE_CORELIB compilation constant.
